drivers/sensor/qdec_nxp_s32/qdec_nxp_s32.c

```diff
@@ -0,0 +1,424 @@
+		config->emios_inst, config->emios_channels[EMIOS_CCW_CH_IDX], (uint32_t)0x1U);
+
+	Emios_Icu_Ip_SetMaxCounterValue(config->emios_inst,
+		config->emios_channels[EMIOS_CW_CH_IDX],
+		EMIOS_ICU_IP_COUNTER_MASK);
+	Emios_Icu_Ip_SetMaxCounterValue(config->emios_inst,
+		config->emios_channels[EMIOS_CCW_CH_IDX],
+		EMIOS_ICU_IP_COUNTER_MASK);
+
+	/* This API sets MCB/EMIOS_ICU_MODE_EDGE_COUNTER mode */
+	Emios_Icu_Ip_EnableEdgeCount(config->emios_inst, config->emios_channels[EMIOS_CW_CH_IDX]);
+	Emios_Icu_Ip_EnableEdgeCount(config->emios_inst, config->emios_channels[EMIOS_CCW_CH_IDX]);
+
+	return 0;
+}
+
+#define EMIOS_NXP_S32_INSTANCE_CHECK(idx, node_id)						\
+	((DT_REG_ADDR(node_id) == IP_EMIOS_##idx##_BASE) ? idx : 0)
+
+#define EMIOS_NXP_S32_GET_INSTANCE(node_id)							\
+	LISTIFY(__DEBRACKET eMIOS_INSTANCE_COUNT, EMIOS_NXP_S32_INSTANCE_CHECK, (|), node_id)
+
+/* LCU Logic Input Configuration */
+#define LogicInputCfg_Common(mux_sel)								   \
+	{											   \
+		.MuxSel = mux_sel,								   \
+		.SwSynMode = LCU_IP_SW_SYNC_IMMEDIATE,						   \
+		.SwValue = LCU_IP_SW_OVERRIDE_LOGIC_LOW,					   \
+	};
+#define LogicInput_Config_Common(n, hw_lc_input_id, logic_input_n_cfg)				   \
+	{											   \
+		.xLogicInputId = {								   \
+			.HwInstId = DT_INST_PROP(n, lcu_inst),					   \
+			.HwLcInputId = hw_lc_input_id,						   \
+		},										   \
+		.pxLcInputConfig = &logic_input_n_cfg,						   \
+	};
+
+/* LCU Logic Output Configuration */
+#define LogicOutputCfg_Common(En_Debug_Mode, Lut_Control, Lut_Rise_Filt, Lut_Fall_Filt)		   \
+	{											   \
+		.EnDebugMode = (boolean)En_Debug_Mode,						   \
+		.LutControl = Lut_Control,							   \
+		.LutRiseFilt = Lut_Rise_Filt,							   \
+		.LutFallFilt = Lut_Fall_Filt,							   \
+		.EnLutDma = (boolean)FALSE,							   \
+		.EnForceDma = (boolean)FALSE,							   \
+		.EnLutInt = (boolean)FALSE,							   \
+		.EnForceInt = (boolean)FALSE,							   \
+		.InvertOutput = (boolean)FALSE,							   \
+		.ForceSignalSel = 0U,								   \
+		.ClearForceMode = LCU_IP_CLEAR_FORCE_SIGNAL_IMMEDIATE,				   \
+		.ForceSyncSel = LCU_IP_SYNC_SEL_INPUT0,						   \
+	};
+#define LogicOutput_Config_Common(n, logic_output_cfg, hw_lc_output_id)				   \
+	{											   \
+		.xLogicOutputId = {								   \
+			.HwInstId = DT_INST_PROP(n, lcu_inst),					   \
+			.HwLcOutputId = hw_lc_output_id,					   \
+			.IntCallback = NULL_PTR,						   \
+		},										   \
+		.pxLcOutputConfig = &logic_output_cfg,						   \
+	};
+
+#define LCU_IP_INIT_CONFIG(n)									\
+	const Lcu_Ip_LogicInputConfigType LogicInput0Cfg =					\
+		LogicInputCfg_Common(LCU_IP_MUX_SEL_LU_IN_0)					\
+	const Lcu_Ip_LogicInputConfigType LogicInput1Cfg =					\
+		LogicInputCfg_Common(LCU_IP_MUX_SEL_LU_IN_1)					\
+	const Lcu_Ip_LogicInputConfigType LogicInput2Cfg =					\
+		LogicInputCfg_Common(LCU_IP_MUX_SEL_LU_OUT_0)					\
+	const Lcu_Ip_LogicInputConfigType LogicInput3Cfg =					\
+		LogicInputCfg_Common(LCU_IP_MUX_SEL_LU_OUT_1)					\
```

Comment:
Will this work for multiple instances? I don't see the instance number `n` being used here

Reply:
For now we are not planning on using multiple motors, but can think of adding this support later on.
Can we defer it and cover it in - https://github.com/zephyrproject-rtos/zephyr/issues/65459

Reply:
I have added another item, to make this driver multi instance, in the TODO list of #65459

Reply:
No, it needs to be addressed in this PR